memreps/memreps.py

```diff
@@ -71,12 +73,44 @@ def find_distiguishing(concept1, concept2, atoms):
             assert (y in concept2) == polarity
             return x, y
 
+def find_maximally_distinguishing(concept_gen, max_concepts=10, atoms_per_concept=1):
+    atom_set = set()
+    # build list of atoms
+    all_concepts = concept_gen()
+    for concept_num, concept in enumerate(all_concepts):
```

Comment:
```suggestion
from itertools import combinations, islice

concepts = fn.take(make_concepts, concept_gen())
pairs = combinations(concepts, 2)
diffs = (c1 ^ c2 for c1, c2 in pairs)
atoms = (set(fn.take(atoms_per_concept, c)) for c in diffs)
atoms = set.union(*atoms)
atoms = sorted(atoms, key=lambda x: sum((x in c) for c in concepts))
return atoms[0], atoms[1]
```

Reply:
looks good except we're trying to find |f(x) - 1/2| where f(x) is the fraction of concepts with membership, no?

Reply:
it seems like we weren't doing the proper reduction in `summarize` either? made that change too